nixos/modules/services/hardware/fancontrol.nix

```diff
@@ -4,42 +4,41 @@ with lib;
 
 let
   cfg = config.hardware.fancontrol;
-  configFile = pkgs.writeText "fan.conf" cfg.config;
-
-in {
+  configFile = pkgs.writeText "fancontrol" cfg.config;
 
+in{
   options.hardware.fancontrol = {
-    enable = mkEnableOption "fancontrol (requires fancontrol.config)";
+    enable = mkEnableOption "software fan control (requires fancontrol.config)";
 
     config = mkOption {
-      type = types.lines;
       default = null;
+      type = types.str;
```

Comment:
`types.lines` is the better choice here. It allows for merging configs.

Reply:
i'm aware
i don't believe this config is one that can be merged by appending (you can't use the same variable twice)
see DEVNAME for example, 2 entries, on 1 line

Reply:
You could merge files for example where you have the same hardware but want to run different temperatures.

Reply:
right!